File: app/src/main/java/com/example/ign_app/networkcalls/IgnApi.java
package com.example.ign_app.networkcalls;

import com.example.ign_app.articlemodel.ArticleApiResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Query;

public interface IgnApi {

    String BASE_URL = "https://ign-apis.herokuapp.com/";

    @Headers("Content-Type: application/json")
    @GET("articles")
    Call<ArticleApiResponse> getArticleJson(@Query("startIndex") int startIndex,
                                            @Query("count") int count);

}

File: app/src/main/java/com/example/ign_app/networkcalls/IgnApi.kt
package com.example.ign_app.networkcalls

import com.example.ign_app.articlemodel.ArticleApiResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query

interface IgnApi {
    @Headers("Content-Type: application/json")
    @GET("articles")
    fun getArticleJson(@Query("startIndex") startIndex : Int,
                       @Query("count") count : Int): Call<ArticleApiResponse?>?

    companion object {
        const val BASE_URL = "https://ign-apis.herokuapp.com/"
    }
}